### PR TITLE
COORDINATIONS: Identifications ProcessImpersonationContextRequest

### PR DESCRIPTION
**Pull request by MichaelOSullivanAnswer**
CLOSES #AB16458

### CODE REVIEW

File: ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs [hunk 1]
```diff
@@ -111,12 +113,57 @@ await ConvertIdentificationRequestToCsvIdentificationRequest(
         public ValueTask<AccessRequest> PersistsImpersonationContextAsync(AccessRequest accessRequest) =>
         TryCatch(async () =>
         {
-            ValidateOnPersistsImpersonationContext(accessRequest);
+            ValidateImpersonationContext(accessRequest);
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs:116]: New line before a return

File: ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs [hunk 2]
```diff
@@ -111,12 +113,58 @@ await ConvertIdentificationRequestToCsvIdentificationRequest(
...
+        TryCatch(async () =>
+        {
+            ValidateImpersonationContext(accessRequest);
+
+            IdentificationRequest identificationRequest =
+                await ConvertCsvIdentificationRequestToIdentificationRequest(
+                    accessRequest.CsvIdentificationRequest);
+
+            AccessRequest convertedAccessRequest = new AccessRequest();
+            string filepath = accessRequest.CsvIdentificationRequest.Filepath;
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs:131]: Is this not working on a zero based index?

File: ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs [hunk 3]
```diff
@@ -111,12 +113,58 @@ await ConvertIdentificationRequestToCsvIdentificationRequest(
...
+            string[] filepathParts = filepath.Split('/');
+            string container = filepathParts[1];
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs:133]: This looks wrong, the file path is everything apart from the container. i.e. /{impersonationContextId}/{inbox}/something.csv

File: ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs (hunk 2 above)
Comment (MichaelOSullivanAnswer) [reply ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs:131]: it depends on if there is a leading '/' in the filepath. I worked from the assumption there is, so the 0 index item in the split array is "".

File: ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+                service.ConvertCsvIdentificationRequestToIdentificationRequest(
+                    inputAccessRequest.CsvIdentificationRequest))
+                        .ReturnsAsync(outputConversionIdentificationRequest);
+
+            this.persistanceOrchestrationServiceMock.Setup(service =>
+                service.RetrieveImpersonationContextByIdAsync(contextId))
+                    .ReturnsAsync(outputPersistanceOrchestrationAccessRequest);
+
+            this.accessOrchestrationServiceMock.Setup(service =>
+                service.ValidateAccessForIdentificationRequestAsync(It.IsAny<AccessRequest>()))
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs:94]: You are not allowed to use `It.IsAny<>()` in logic tests.  You need to setup and verify that the methods are called with the correct values

File: ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+                    .ReturnsAsync(outputOrchestrationAccessRequest);
+
+            this.identificationOrchestrationServiceMock.Setup(service =>
+                service.ProcessIdentificationRequestAsync(It.IsAny<IdentificationRequest>()))
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs:98]: `It.IsAny<>()` not allowed

**Review by JParkesAD — DISMISSED** [review]

File: ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs (hunk 4 above)
Comment (MichaelOSullivanAnswer) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs]: This has been updated

File: ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs (hunk 5 above)
Comment (MichaelOSullivanAnswer) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ProcessImpersonationContextRequest.cs]: This has been updated

File: ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ExtractFromFilepath.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            string contextIdString = contextId.ToString();
+            string randomString = GetRandomString();
+            string container = GetRandomString();
+            string fileName = randomString;
+            string fileExtension = ".csv";
+            string filepath = $"/{container}/{contextIdString}/outbox/subdirectory/{fileName}{fileExtension}";
+            string inputFilepath = filepath;
+            string landingFolderName = GetRandomString();
+            string pickupFolderName = GetRandomString();
+            string errorFolderName = GetRandomString();
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ExtractFromFilepath.cs:33]:
```suggestion
            string landingFolderName = GetRandomString();
            string pickupFolderName = GetRandomString();
            string errorFolderName = GetRandomString();
            string filepath = $"/{container}/{contextIdString}/{landingFolderName}/subdirectory/{fileName}{fileExtension}";
            string inputFilepath = filepath;
```

Comment (MichaelOSullivanAnswer) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.ExtractFromFilepath.cs]: Fixed

**Review by cjdutoit — DISMISSED** [review]

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Program.cs [hunk 7]
```diff
@@ -211,8 +212,12 @@ private static void AddOrchestrationServices(IServiceCollection services)
             services.AddTransient<ICsvIdentificationRequestService, CsvIdentificationRequestService>();
         }
 
-        private static void AddCoordinationServices(IServiceCollection services)
+        private static void AddCoordinationServices(IServiceCollection services, IConfiguration configuration)
         {
+            ProjectStorageConfiguration projectStorageConfiguration = configuration
+                .GetSection("projectStorageConfiguration")
+                    .Get<ProjectStorageConfiguration>();
```
Comment (cjdutoit) [inline ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Program.cs:219]:
```suggestion
            ProjectStorageConfiguration projectStorageConfiguration = configuration
                .GetSection("projectStorageConfiguration")
                    .Get<ProjectStorageConfiguration>();

            services.AddSingleton(projectStorageConfiguration);
```

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs [hunk 8]
```diff
@@ -203,8 +204,12 @@ private static void AddOrchestrationServices(IServiceCollection services)
             services.AddTransient<ICsvIdentificationRequestService, CsvIdentificationRequestService>();
         }
 
-        private static void AddCoordinationServices(IServiceCollection services)
+        private static void AddCoordinationServices(IServiceCollection services, IConfiguration configuration)
         {
+            ProjectStorageConfiguration projectStorageConfiguration = configuration
+                .GetSection("projectStorageConfiguration")
+                    .Get<ProjectStorageConfiguration>();
```
Comment (cjdutoit) [inline ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs:211]:
```suggestion
            ProjectStorageConfiguration projectStorageConfiguration = configuration
                .GetSection("projectStorageConfiguration")
                    .Get<ProjectStorageConfiguration>();

            services.AddSingleton(projectStorageConfiguration);
```

**Pull request merged**